sootup.jimple.frontend/src/main/java/sootup/jimple/frontend/JavaCodeBuilder.java

```diff
@@ -0,0 +1,180 @@
+            + "null,"
+            + "Collections.emptySet(),"
+            + "null,"
+            + "Collections.emptySet(),"
+            + "Collections.singleton(dummyMainMethod),"
+            + "NoPositionInformation.getInstance(),"
+            + "EnumSet.of(ClassModifier.PUBLIC),"
+            + "Collections.emptyList(),"
+            + "Collections.emptyList(),"
+            + "Collections.emptyList()),"
+            + "SourceType.Application);";
+    javaCodeObjects.add(classString);
+  }
+
+  public void createMethod() {
+    String methodString =
+        "JavaSootMethod dummyMainMethod = new JavaSootMethod("
+            + "new OverridingBodySource(methodSignature, body),"
+            + "methodSignature,"
+            + "EnumSet.of(MethodModifier.PUBLIC, MethodModifier.STATIC),"
+            + "Collections.emptyList(),"
+            + "Collections.emptyList(),"
+            + "NoPositionInformation.getInstance());";
+    javaCodeObjects.add(methodString);
+  }
+
+  public void createStmtGraph() {
+    javaCodeObjects.add("MutableStmtGraph stmtGraph = bodyBuilder.getStmtGraph();");
+    javaCodeObjects.add(String.format("stmtGraph.setStartingStmt(%s);", javaCodeObjects.get(0)));
+  }
+
+  public void initJavaCode() {
+    javaCodeObjects.add("Body.BodyBuilder bodyBuilder = Body.builder()");
+    javaCodeObjects.add(
+        String.format(
+            "JavaIdentifierFactory %s = JavaIdentifierFactory.getInstance();", "factory"));
+    javaCodeObjects.add(
+        String.format(
+            "StmtPositionInfo %s = StmtPositionInfo.getNoStmtPositionInfo();",
+            "noStmtPositionInfo"));
+  }
+
+  public void addThisRef(Local leftOp, Type classType) {
+    javaCodeObjects.add(
+        String.format(
+            "IdentityRef %s = JavaJimple.newThisRef(factory.getClassType(\"%s\"));",
+            leftOp.getName(), classType));
+  }
+
+  public void addLocal(Local value) {
+    javaCodeObjects.add(
+        String.format(
+            "Local %s = JavaJimple.newLocal(\"%s\", factory.getClassType(\"%s\"));",
+            value, value.getName(), value.getType()));
+  }
+
+  public void addAssignment(JAssignStmt stmt) {
+    String typeString = "";
+    if (stmt.getRightOp() instanceof IntConstant) {
+      typeString =
+          String.format(
+              "IntConstant.getInstance(%s)", ((IntConstant) stmt.getRightOp()).getValue());
+    } else if (stmt.getRightOp() instanceof Local) {
+      typeString =
+          String.format(
+              "JavaJimple.newLocal(\"%s\", factory.getClassType(\"%s\"))",
+              ((Local) stmt.getRightOp()).getName(), stmt.getRightOp().getType());
+    }
+    javaCodeObjects.add(
+        String.format(
+            "Stmt %s = JavaJimple.newAssignment(JavaJimple.newLocal(\"%s\", factory.classType(\"%s\")), %s , noStmtPositionInfo)",
+            stmt.getLeftOp().toString(),
+            stmt.getLeftOp().toString(),
+            stmt.getType().toString(),
+            typeString));
+  }
```

Comment:
don't forget we have more of those visitors like the ValueVisitor e.g. for Operands